lib/logger/lib/logger.ex

```diff
@@ -990,15 +994,17 @@ defmodule Logger do
   end
 
   defp macro_log(level, data, metadata, caller) do
-    caller =
-      compile_time_application_and_file(caller) ++
-        case caller do
-          %{module: module, function: {fun, arity}, line: line} ->
-            [mfa: {module, fun, arity}, line: line]
-
-          _ ->
-            []
-        end
+    [{:file, file} | _] = application_and_file = compile_time_application_and_file(caller)
+
+    {caller, location} =
+      case caller do
+        %{module: module, function: {fun, arity}, line: line} ->
+          {application_and_file ++ [mfa: {module, fun, arity}, line: line],
```

Comment:
We shokd not include mfa and file and stuff in the caller if it is already on location. :)

Reply:
Updated!